test/powershell/Modules/Microsoft.PowerShell.Management/Resolve-Path.Tests.ps1

```diff
@@ -48,8 +48,64 @@ Describe "Resolve-Path returns proper path" -Tag "CI" {
             Pop-Location
         }
     }
-    It 'Resolve-Path should support user specified base paths' {
-        $Expected = Join-Path -Path .\ -ChildPath fakeroot
-        Resolve-Path -Path $fakeRoot -RelativeBasePath $testRoot | Should -BeExactly $Expected
+    It 'Resolve-Path RelativeBasePath should handle <Scenario>' -TestCases @(
+        @{
+            Scenario = "Absolute Path, Absolute ReleativeBasePath"
+            Path     = $root
+            Basepath = $testRoot
+            Expected = $root, ".$([System.IO.Path]::DirectorySeparatorChar)fakeroot"
+            CD       = $null
+        }
+        @{
+            Scenario = "Relative Path, Absolute ReleativeBasePath"
+            Path     = ".$([System.IO.Path]::DirectorySeparatorChar)fakeroot"
+            Basepath = $testRoot
+            Expected = $root, ".$([System.IO.Path]::DirectorySeparatorChar)fakeroot"
+            CD       = $null
+        }
+        @{
+            Scenario = "Relative Path, Relative ReleativeBasePath"
+            Path     = ".$([System.IO.Path]::DirectorySeparatorChar)fakeroot"
+            Basepath = ".$([System.IO.Path]::DirectorySeparatorChar)"
+            Expected = $root, ".$([System.IO.Path]::DirectorySeparatorChar)fakeroot"
+            CD       = $testRoot
+        }
+        @{
+            Scenario = "Invalid Path, Absolute ReleativeBasePath"
+            Path     = Join-Path $testRoot ThisPathDoesNotExist
+            Basepath = $root
+            Expected = $null
+            CD       = $null
+        }
+        @{
+            Scenario = "Invalid Path, Invalid ReleativeBasePath"
+            Path     = Join-Path $testRoot ThisPathDoesNotExist
+            Basepath = Join-Path $testRoot ThisPathDoesNotExist
+            Expected = $null
+            CD       = $null
+        }
+    ) -Test {
+        param($Path, $BasePath, $Expected, $CD)
+        
+        if ($null -eq $Expected)
+        {
+            {Resolve-Path -Path $Path -RelativeBasePath $BasePath -ErrorAction Stop} | Should -Throw
```

Comment:
Is the ErrorId consistent such that it could be checked `Should -Throw -ErrorId "..."`